### PR TITLE
feat: #96 create ErrorBoundary 

### PR DESCRIPTION
**Pull request by chelovek-zeml**

### CODE REVIEW

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, {Component} from "react";
+import {Page404} from "src/pages/page404/Page404";
+
+interface Props {
+  children: React.ReactNode;
```
Comment (tritonJS826) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:5]: use PropsWithChildren

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+interface State {
+  hasError: boolean;
+}
+
+class ErrorBoundary extends Component<Props, State> {
```
Comment (tritonJS826) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:12]: let's use functional component. Investigate does possible for error boundaries for react 18

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]
Please check naming of PR and issue

**Comment by Ekaterina1994** [comment]
> Please check naming of PR and issue

I created a new issue with your task. PLease rename PR and last commit with new number. After it put issue #57 into backlog (his task about handle requests errors).

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 1 above)
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:5]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 2 above)
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:12]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, {Component} from "react";
+import {Page404} from "src/pages/page404/Page404";
+
+interface Props {
+  children: React.ReactNode;
+}
```
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:6]: Done.

File: mw-webapp/src/App.tsx [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 
 function App() {
   return (
     <>
       <Header />
-      <Router />
+      <ErrorBoundary>
+        <Router />
+      </ErrorBoundary>
     </>
```
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/App.tsx:12]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    this.state = {hasError: false};
+  }
+
+  public componentDidCatch(): void {
+    this.setState({hasError: true});
+  }
+
+  public render(): React.ReactNode {
+    if (this.state.hasError) {
+      return <Page404 />;
```
Comment (tritonJS826) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:38]: why this logic? if you have some problem with rendering it does not mean that page not found

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+ *   They are React components that catch JavaScript errors anywhere in their child component tree,
+ * log those errors, and display a fallback UI instead of the component tree that crashed.
+
+     It’s important to note that error boundaries catch errors during rendering, in lifecycle methods,
+   and constructors of the whole tree below them. However, error boundaries do not catch errors for:
+
+   Event handlers (for that, you need to use regular try/catch)
+   Asynchronous code (e.g., setTimeout or requestAnimationFrame callbacks)
+   Server-side rendering
+   Errors thrown in the error boundary itself (rather than its children).
```
Comment (tritonJS826) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:14]: could you provide an example of error which will be catched by this component?

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+  public render(): React.ReactNode {
+    if (this.state.hasError) {
+      return <Page404 />;
+    }
+    return this.props.children;
+  }
+
+}
+
+export default ErrorBoundary;
```
Comment (tritonJS826) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:45]: we dont use default exports. Please adjust linter to forbid all default exports in project

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tritonJS826) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:19]: https://blog.logrocket.com/using-react-children-prop-with-typescript/

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/App.tsx [hunk 8]
```diff
@@ -1,11 +1,14 @@
+import ErrorBoundary from "src/component/errorBoundary/ErrorBoundary";
 import {Header} from "src/component/header/Header";
 import {Router} from "src/router/Router";
 
 function App() {
```
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/App.tsx:5]: Done.

File: mw-webapp/src/App.tsx [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
   return (
     <>
```
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/App.tsx:7]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 6 above)
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:14]: Added. Will this example work?

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:19]: I read this. PropsWithChildren receives any types + ReactNode, that is, it is more universal. Did I understand correctly?

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 7 above)
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:44]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+class ErrorBoundary extends Component<PropsWithChildren, State> {
+
+  constructor(props: PropsWithChildren) {
+    super(props);
+    this.state = {hasError: false};
+  }
+
+  public componentDidCatch(): void {
```
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:31]: an error appears without the public attribute
35:3  error  Missing accessibility modifier on method definition render  @typescript-eslint/explicit-member-accessibility

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 5 above)
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:37]: Done.

File: mw-webapp/src/App.tsx [hunk 11]
```diff
@@ -1,13 +1,14 @@
...
+    <ErrorBoundary>
       <Header />
       <Router />
-    </>
+    </ErrorBoundary>
   );
-}
+};
 
 export default App;
```
Comment (Ekaterina1994) [inline mw-webapp/src/App.tsx:14]: Don't use default export

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (Ekaterina1994) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:30]: Add docs

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

**Review by Ekaterina1994 — COMMENTED** [review]
Resolve conflicts before rereview)

File: mw-webapp/src/App.tsx (hunk 11 above)
Comment (chelovek-zeml) [reply mw-webapp/src/App.tsx:14]: Done.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:30]: Add method description? Or should I add something to the documentation?

[the remark this answers is not in the document]
Comment (chelovek-zeml) [reply mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:30]: Do not quite understand.

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+  public render(): React.ReactNode {
+    if (this.state.hasError) {
+      return (<h1>
+        Something went wrong.
+      </h1>);
+    }
+    return this.props.children;
+  }
+
+}
```
Comment (Howl404) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:43]:
```suggestion
  public render(): React.ReactNode {
    if (this.state.hasError) {

      return (
        <h1>
          Something went wrong.
        </h1>
      );
    }

    return this.props.children;
  }

}
```

**Review by Howl404 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    this.setState({hasError: true});
+  }
+
+  public render(): React.ReactNode {
+    if (this.state.hasError) {
+    // You can render the fallback UI.
+      return (<h1>
+        Something went wrong.
+      </h1>);
+    }
```
Comment (Ekaterina1994) [inline mw-webapp/src/component/errorBoundary/ErrorBoundary.tsx:41]: check formatting

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**